src/feat/wave-reader.h

```diff
@@ -171,6 +172,11 @@ class WaveHolder {
     t_.Swap(&(other->t_));
   }
 
+  bool ExtractRange(const WaveHolder &other, const std::string &range) {
+    KALDI_ERR << "ExtractRange is not defined for this type of holder.";
```

Comment:
I had a look into it, and I think I'd rather leave it as is.  It feels like the use of hacks like this is working against the grain of the language, and would be hard to debug.